torch/csrc/distributed/rpc/ProcessGroupAgent.cpp

```diff
@@ -117,14 +125,9 @@ void ProcessGroupAgent::sync() {
 }
 
 std::shared_ptr<FutureMessage> ProcessGroupAgent::send(
-    const std::string& to, Message&& message) {
-
-  auto dstRankIter = nameMap_.find(to);
-  TORCH_CHECK(dstRankIter != nameMap_.end(), "Unknown destination worker ", to);
-
-  const int dstRank = dstRankIter -> second;
-  TORCH_CHECK(dstRank != pg_->getRank(), "ProcessGroupAgent does not support "
-    "making RPC calls to self.")
+    worker_id_t to, Message&& message) {
+  TORCH_CHECK(to != (worker_id_t)pg_->getRank(),
```

Comment:
Should we keep checking `to < pg_->world_size_`?

Reply:
Yes, let me add that